src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -166,27 +164,22 @@ public synchronized boolean open() throws CouchbaseLiteException {
             throw new IllegalStateException(msg);
         }
 
-        boolean isOpen = false;
-        Throwable error = null;
+        boolean isOpenSuccess = false;
         try {
+            // Open database:
+            storageEngine.open(path);
+            // Try to decrypt or access the database:
             SymmetricKey encryptionKey = delegate.getEncryptionKey();
-            if (encryptionKey != null && !storageEngine.supportEncryption()) {
-                throw new CouchbaseLiteException(
-                        "SQLiteStore: encryption not available (app not built with SQLCipher)",
-                        Status.NOT_IMPLEMENTED);
-            }
-            String rawKey = encryptionKey != null ? encryptionKey.getHexData() : null;
-            isOpen = storageEngine.open(path, rawKey);
+            decrypt(encryptionKey);
```

Comment:
Yes. The decrypt method has a logic to validate whether or not the encryption is used. Also this follows the current iOS implementation.